pkg/core/interops.go

```diff
@@ -74,28 +85,33 @@ var systemInterops = []interop.Function{
+		RequiredFlags: smartcontract.AllowModifyStates, ParamCount: 2},
+	{Name: "System.Storage.Find", Func: storageFind, Price: 1000000, RequiredFlags: smartcontract.AllowStates, ParamCount: 2},
+	{Name: "System.Storage.Get", Func: storageGet, Price: 1000000, RequiredFlags: smartcontract.AllowStates, ParamCount: 2},
 	{Name: "System.Storage.GetContext", Func: storageGetContext, Price: 400, RequiredFlags: smartcontract.AllowStates},
 	{Name: "System.Storage.GetReadOnlyContext", Func: storageGetReadOnlyContext, Price: 400, RequiredFlags: smartcontract.AllowStates},
-	{Name: "System.Storage.Put", Func: storagePut, Price: 0, RequiredFlags: smartcontract.AllowModifyStates}, // These don't have static price in C# code.
-	{Name: "System.Storage.PutEx", Func: storagePutEx, Price: 0, RequiredFlags: smartcontract.AllowModifyStates},
-	{Name: "System.Storage.AsReadOnly", Func: storageContextAsReadOnly, Price: 400, RequiredFlags: smartcontract.AllowStates},
+	{Name: "System.Storage.Put", Func: storagePut, Price: 0, RequiredFlags: smartcontract.AllowModifyStates,
+		ParamCount: 3}, // These don't have static price in C# code.
+	{Name: "System.Storage.PutEx", Func: storagePutEx, Price: 0, RequiredFlags: smartcontract.AllowModifyStates, ParamCount: 4},
+	{Name: "System.Storage.AsReadOnly", Func: storageContextAsReadOnly, Price: 400,
+		RequiredFlags: smartcontract.AllowStates, ParamCount: 1},
 }
 
 var neoInterops = []interop.Function{
-	{Name: "Neo.Crypto.VerifyWithECDsaSecp256r1", Func: crypto.ECDSASecp256r1Verify, Price: crypto.ECDSAVerifyPrice},
-	{Name: "Neo.Crypto.VerifyWithECDsaSecp256k1", Func: crypto.ECDSASecp256k1Verify, Price: crypto.ECDSAVerifyPrice},
-	{Name: "Neo.Crypto.CheckMultisigWithECDsaSecp256r1", Func: crypto.ECDSASecp256r1CheckMultisig, Price: 0},
-	{Name: "Neo.Crypto.CheckMultisigWithECDsaSecp256k1", Func: crypto.ECDSASecp256k1CheckMultisig, Price: 0},
-	{Name: "Neo.Crypto.SHA256", Func: crypto.Sha256, Price: 1000000},
-	{Name: "Neo.Crypto.RIPEMD160", Func: crypto.RipeMD160, Price: 1000000},
-	{Name: "Neo.Native.Call", Func: native.Call, Price: 0},
-	{Name: "Neo.Native.Deploy", Func: native.Deploy, Price: 0, RequiredFlags: smartcontract.AllowModifyStates},
+	{Name: "Neo.Crypto.VerifyWithECDsaSecp256r1", Func: crypto.ECDSASecp256r1Verify,
+		Price: crypto.ECDSAVerifyPrice, ParamCount: 2},
+	{Name: "Neo.Crypto.VerifyWithECDsaSecp256k1", Func: crypto.ECDSASecp256k1Verify,
+		Price: crypto.ECDSAVerifyPrice, ParamCount: 2},
+	{Name: "Neo.Crypto.CheckMultisigWithECDsaSecp256r1", Func: crypto.ECDSASecp256r1CheckMultisig, Price: 0, ParamCount: 1},
+	{Name: "Neo.Crypto.CheckMultisigWithECDsaSecp256k1", Func: crypto.ECDSASecp256k1CheckMultisig, Price: 0, ParamCount: 1},
+	{Name: "Neo.Crypto.SHA256", Func: crypto.Sha256, Price: 1000000, ParamCount: 1},
+	{Name: "Neo.Crypto.RIPEMD160", Func: crypto.RipeMD160, Price: 1000000, ParamCount: 1},
+	{Name: "Neo.Native.Call", Func: native.Call, Price: 0, ParamCount: 3},
```

Comment:
And that's an interesting thing, because to match C# implementation this interop should have 1 parameter (see https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Native.cs#L28) and after that there's `Invoke` method which pops from the stack the rest of the arguments. Perhaps, it would be better to set `ParamCount` to `1` here although that's not exactly true.

On the other hand, it's not so important because callbacks are not allowed for this interop. Perhaps, we shouldn't even set the number of parameters for those interops which not allow callbacks.

Reply:
Nice catch! It is important for creating callbacks.